docs(sessions): :memo: add text for big picture (#99)
rostools/git4cats: sessions/introduction.qmd

    pull requests by making comments and suggestions.
-   Accepting contributions by merging changes back into the main branch
    of the project.

### Why we want that {.unnumbered}

Working well together on a common project isn't easy nor a thing we can
naturally do. It requires a lot of skill, knowledge, and actually
applying these practices often in real settings. Above all though, it
requires *explicitly* agreeing and committing to working a certain with
your team members. Agreeing to work better together can be easy. Knowing
*how exactly* to work together is hard, *explicitly* agreeing on those
is harder, and actually committing to those practices is the hardest.

Too often, we just say "oh, it's just a small thing" or "I have to do
this quickly" or "It's too much learning or change for me, I'll just do
it my way". But, if we don't agree and actually commit to working better
together, we'll end up with a mess of a project, with no one knowing
what's going on, who's doing what, and what the current state of the
project is. By following the same agreed-upon collaborative practices,
you as a team can focus on the content, on understanding what others are

> contributing to the project, and providing relevant feedback directly to

Reviewer: ```suggestion
contributing to the project, and providing relevant feedback directly
```

where it is needed.

So, to help with *knowing how exactly to work together*, here are the
reasons why we use the tools and practices we do based on our
experiences and research into what are the most optimal and effective

> ways to collaborate together as a team:

Reviewer: Not sure this fits the "why we want that" header? Should this and the list below be under "how we'll get there"? 
Also not entirely sure what is meant by this sentence actually 🤔 😅

-   We use Git and GitHub because they are both incredibly popular tools
    for collaboration.

> -   We use the GitHub flow (using Git branches and pull requests)
>     because it allows each team member to work on their own part of the
>     project without interfering with others' work and makes it easy to
>     get feedback and reviews from others when that work is ready to be
>     added back to the main content of the project.

Reviewer: ```suggestion
-   We use the GitHub flow (using branches and pull requests)
    to let team members work on their own part of the
    project without interfering with others' work and easily
    get feedback before adding this work to the project.
```

> -   We use "atomic" commits, which means the commit contains a single
>     conceptual change (e.g. only changes a typo), so that it is easier
>     for the reviewers (our team members) to understand what we changed
>     and why.

Reviewer: ```suggestion
-   We use "atomic" commits, i.e., each commit contains a single
    conceptual change (e.g. only adds or fixes one thing), so that it is easier
    for our team members to understand what we changed
    and why.
```

-   We use the Conventional Commits standard to write commit messages
    because it ensures that messages are consistent between team members

>     and it more clearly communicates what was done in a commit.

Reviewer: ```suggestion
    and it more clearly communicates what was done.
```

> -   We use GitHub pull requests to be able to directly comment on and
>     give feedback to changes made by other team members so that review,
>     discussions, and decisions can be visible to everyone on the team.

Reviewer: ```suggestion
-   We use GitHub pull requests to directly review and comment on
    changes made by team members so that feedback,
    discussions, and decisions is visible to everyone on the team.
```

> For more reasons on why we use these tools and practices, see the

Reviewer: Good example of where drawing the line between a rostools `decisions` repo and a Seedcase `decisions` repo will be difficult (I know I'm the one who suggested it, but this is an example of why it might be difficult to split up, even though I would like it to be)

> [decision posts](https://decisions.seedcase-project.org) of our sibling

Reviewer: ```suggestion
[decision posts](https://decisions.seedcase-project.org) of our parent
```
Orrr mother? haha

the [Seedcase Project](https://seedcase-project.org).

While the reasons for these choices are grounded in research,
experience, and common practice in industry, there are costs related to
these choices. The main one is that at this current time, few people who

> work in researcher use these practices. Part of the aim of this course

Reviewer: ```suggestion
work in research use these practices. Part of the aim of this course
```

> is to spread awareness and education so that in the future more
> researchers and those in academia can be more familiar with the tools
> and practices that we will cover in this course.

Reviewer: ```suggestion
is to spread awareness and educate researchers so that in the future
so the tools and practices covered in this course will become more common in academia.
```

> ### How we will get there {.unnumbered}

Author: ```suggestion
### What we will do in this course {.unnumbered}
```

For this course, how you will learn to use more effective collaborative
practices within a team setting is by:

-   Participating in live-coding sessions where we will demonstrate how
    to use Git and GitHub for collaboration.

> -   Discuss with your neighbour about what makes a collaborative

Reviewer: ```suggestion
-   Discussing with your neighbour what makes a collaborative
```

    workflow effective.
-   Working on hands-on exercises where you will practice using Git and

>     GitHub on making files, editing files, making pull requests,

Reviewer: ```suggestion
    GitHub by making and editing files, making pull requests,
```

    reviewing pull requests, and merging pull requests.

> -   Work with at least one other person on a project, where you start

Reviewer: ```suggestion
-   Working with at least one other person on a project, where you start
```

    from the beginning and apply all the practices we've learned in the
    course.

> ### What it will look like {.unnumbered}

Reviewer: I like having this section to show form the beginning what we're working towards 👍 🌟
